### DISCUSSION TITLE
Python / VSCode: Learn if debugger is currently attached to a process

### DISCUSSION THREAD

**Discussion by LukasHeller** [microsoft/debugpy#1526] [Q&A] [answered]
I am using debugby to attach (and de-attach) a debugger in VS Code. I would like to let my Python script know, at runtime, if the debugger is currently attached. Is this possible?

Thanks a lot for your help, Best regards, Lukas

**Comment by int19h** [comment, marked as answer]
If you can successfully `import debugpy` (or better yet check `sys.modules` so as to not import it if it's not there), and then `debugpy.is_client_connected()` returns True, then there is a client attached.

**Reply by LukasHeller** [reply]
Thanks a lot! I had missed this.

On another note, is it possible to use "lauch.json" to forward configuration strings? I am using environment variables to do so in "launch" mode (see below), but this option is not possible in "attach" mode.

    "configurations": [
        {
            "name": "Python: Current File",
            "type": "debugpy",
            "request": "launch",
            "program": "${file}",
            "console": "integratedTerminal",
            "justMyCode": true,
            "env": {
                "VSCODE_DEBUG_MODE": "true",
            }
        },
